### PR TITLE
[FLINK-15648][k8s] Introduce CPU limit support for JM/TM Container

### PR DESCRIPTION
**Pull request by zhengcanbin**
## What is the purpose of the change

Currently, we use `kubernetes.jobmanager(taskmanager).cpu` to configure CPU request and CPU limit at the same time for the JM/TM Container; given that people are asking for different request and limit values for the sake of elasticity, this PR proposes to separate these two configurations, people can set the CPU limit separately via the new config options `kubernetes.jobmanager(taskmanager).cpu.limit`.

## Brief change log

  - Introduce a new config option `kubernetes.jobmanager.cpu.limit`
  - Introduce a new config option `kubernetes.taskmanager.cpu.limit`

## Verifying this change

This change added unit tests and could be covered by existing tests.

## Open questions

1. K8s allows us to set CPU request or limit to zero for a Container. After this PR, 
- JobManager CPU limit allows zero value
- TaskManager CPU limit allows zero value
- JobManager CPU request allows zero value
- But TaskManager CPU request does not allow zero value, this is somewhat inconsistent with K8s or the JobManager side, at the moment I think it is a trivial problem, but we can support it if necessary; welcome any feedback on this issue.

2. For the existing config option of `kubernetes.jobmanager.cpu` and `kubernetes.taskmanager.cpu`, actually they are used for specifying hard CPU request, therefore, I tend to append a postfix `.request` to them. Looking forward to your thoughts.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java [hunk 1]
```diff
@@ -122,19 +123,35 @@ public static String getDeploymentName(String clusterId) {
 	 * Get resource requirements from memory and cpu.
 	 *
 	 * @param mem Memory in mb.
-	 * @param cpu cpu.
+	 * @param cpuRequest cpu request.
```
Comment (kylemeow) [inline flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java:126]: Hi Canbin, thanks for your contribution. Not meant to be a grammar police, however, I have seen multiple 'cpu' occurrences in the description and error messages. For standardization, could you please capitalize it as "CPU"? 

Also please pay attention to its plural form, like "number of cpu", as it should be "number of CPUs" in the description text.

Comment (zhengcanbin) [reply flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java]: > For standardization, could you please capitalize it as "CPU"?
Thanks for pointing out this, the new commit fix it.

> Also please pay attention to its plural form, like "number of cpu", as it should be "number of CPUs" in the description text.
They are not introduced by this PR and indeed are not in the scope.

**Comment by zhengcanbin** [comment]
@TisonKun @tillrohrmann @wangyang0918 Could you help review this PR? I have left two open questions there and looking forward to your thoughts.

**Comment by zhengcanbin** [comment]
Hi @xintongsong. Could you help review this PR?

**Comment by xintongsong** [comment]
Thanks for inviting me, @zhengcanbin.
I'm not entirely sure about the proposed approach. Sorry for haven't bringing this up earlier. I'll comment my concerns on the jira ticket.

**Comment by tillrohrmann** [comment]
Shall we close this PR until we have reached agreement on the JIRA ticket @zhengcanbin?

**Comment by zhengcanbin** [comment]
> Shall we close this PR until we have reached agreement on the JIRA ticket @zhengcanbin?

Sure!

**Pull request closed without merging**